gtsam/linear/iterative.h

```diff
@@ -59,19 +59,14 @@ namespace gtsam {
     /** Access b vector */
     const Vector& b() const { return b_; }
 
-    /** Apply operator A'*e */
-    Vector operator^(const Vector& e) const {
-      return A_ ^ e;
-    }
-
     /**
      * Print with optional string
      */
     void print (const std::string& s = "System") const;
 
     /** gradient of objective function 0.5*|Ax-b_|^2 at x = A_'*(Ax-b_) */
     Vector gradient(const Vector& x) const {
-      return A() ^ (A() * x - b());
+      return A().transpose() * (A() * x - b());
```

Comment:
This might have to be ^ for any "System" to work.

Reply:
Unfortunately, we can't do that because `A()` and `A() * x - b()` both produce Eigen types.